main.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable
+  def my_each
+    index = 0
+
+    while index < size
+      yield self[index]
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    item = 0
+    index = 0
+
+    while index < size
+      yield self[item], index
+      index += 1
+      item += 1
+    end
+  end
+
+  def my_select
+    arr = []
+
+    my_each do |item|
+      arr.push(item) if yield item
+    end
+    arr
+  end
+
+  def my_all?
+    bool = true
+
+    my_each do |item|
+      bool = false unless yield item
+    end
+    bool
+  end
+
+  def my_any?
+    bool = false
+
+    my_each do |item|
+      bool = true if yield item
+    end
+    bool
+  end
+
+  def my_none?
+    bool = true
+
+    my_each do |item|
+      bool = false if yield item
+    end
+    bool
+  end
```

Comment:
- [ ] should have working `my_none?` **(covered by automated tests)** identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains no truthy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_none?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_none?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_none?(8)).`